src/backend/distributed/planner/distributed_planner.c

```diff
@@ -158,7 +158,10 @@ distributed_planner(Query *parse,
 	bool fastPathRouterQuery = false;
 	Node *distributionKeyValue = NULL;
 
-	AttributeQueryIfAnnotated(query_string, parse->commandType);
+	if (query_string != NULL)
```

Comment:
Why is this `if` needed? Afaict `AttributeQueryIfAnnotated` already checks for null internally. I guess maybe this if would not be needed if it's `NULL` check was moved completely to the top of the function. Any reason not to do that?